Network/DNS/Lookup.hs

```diff
@@ -429,3 +430,26 @@ lookupSRV rlv dom = do
", rrtype = TYPE35, rrclass = 1, rrttl = 2222, rdata = \# 56 000100010150094532552b656d61696c0005656d61696c0c7a6f6e657472616e73666572026d650c7a6f6e657472616e73666572026d6500},ResourceRecord {rrname = "email.zonetransfer.me.", rrtype = A, rrclass = 1, rrttl = 7200, rdata = 74.125.206.26},ResourceRecord {rrname = "home.zonetransfer.me.", rrtype = A, rrclass = 1, rrttl = 7200, rdata = 127.0.0.1},ResourceRecord {rrname = "Info.zonetransfer.me.", rrtype = TXT, rrclass = 1, rrttl = 7200, rdata = ZoneTransfer.me service provided by Robin Wood - robin@digi.ninja. See http://digi.ninja/projects/zonetransferme.php for more information.},ResourceRecord {rrname = "internal.zonetransfer.me.", rrtype = NS, rrclass = 1, rrttl = 300, rdata = intns1.zonetransfer.me.},ResourceRecord {rrname = "internal.zonetransfer.me.", rrtype = NS, rrclass = 1, rrttl = 300, rdata = intns2.zonetransfer.me.},ResourceRecord {rrname = "intns1.zonetransfer.me.", rrtype = A, rrclass = 1, rrttl = 300, rdata = 81.4.108.41},ResourceRecord {rrname = "intns2.zonetransfer.me.", rrtype = A, rrclass = 1, rrttl = 300, rdata = 167.88.42.94},ResourceRecord {rrname = "office.zonetransfer.me.", rrtype = A, rrclass = 1, rrttl = 7200, rdata = 4.23.39.254},ResourceRecord {rrname = "ipv6actnow.org.zonetransfer.me.", rrtype = AAAA, rrclass = 1, rrttl = 7200, rdata = 2001:67c:2e8:11::c100:1332},ResourceRecord {rrname = "owa.zonetransfer.me.", rrtype = A, rrclass = 1, rrttl = 7200, rdata = 207.46.197.32},ResourceRecord {rrname = "robinwood.zonetransfer.me.", rrtype = TXT, rrclass = 1, rrttl = 302, rdata = Robin Wood},ResourceRecord {rrname = "rp.zonetransfer.me.", rrtype = TYPE17, rrclass = 1, rrttl = 321, rdata = \# 50 05726f62696e0c7a6f6e657472616e73666572026d650009726f62696e776f6f640c7a6f6e657472616e73666572026d6500},ResourceRecord {rrname = "sip.zonetransfer.me.", rrtype = TYPE35, rrclass = 1, rrttl = 3333, rdata = \# 59 000200030150074532552b7369702b215e2e2a24217369703a637573746f6d65722d73657276696365407a6f6e657472616e736665722e6d652100},ResourceRecord {rrname = "sqli.zonetransfer.me.", rrtype = TXT, rrclass = 1, rrttl = 300, rdata = ' or 1=1 --},ResourceRecord {rrname = "sshock.zonetransfer.me.", rrtype = TXT, rrclass = 1, rrttl = 7200, rdata = () { :]}; echo ShellShocked},ResourceRecord {rrname = "staging.zonetransfer.me.", rrtype = CNAME, rrclass = 1, rrttl = 7200, rdata = www.sydneyoperahouse.com.},ResourceRecord {rrname = "alltcpportsopen.firewall.test.zonetransfer.me.", rrtype = A, rrclass = 1, rrttl = 301, rdata = 127.0.0.1},ResourceRecord {rrname = "testing.zonetransfer.me.", rrtype = CNAME, rrclass = 1, rrttl = 301, rdata = www.zonetransfer.me.},ResourceRecord {rrname = "vpn.zonetransfer.me.", rrtype = A, rrclass = 1, rrttl = 4000, rdata = 174.36.59.154},ResourceRecord {rrname = "www.zonetransfer.me.", rrtype = A, rrclass = 1, rrttl = 7200, rdata = 5.196.105.14},ResourceRecord {rrname = "xss.zonetransfer.me.", rrtype = TXT, rrclass = 1, rrttl = 300, rdata = '><script>alert('Boo')</script>}]
```

Comment:
This example is much too wide, and hardcodes volatile data. Any test of AXFR should be in a test module, rather than be a "doctest". Ideally we could spin up a test server to talk to, so we're not relying on stability of data on the public internet.

Reply:
Agreed. I did not intend to get it on master, but wanted to have something to start the discussion.